frontend/src/Judge/pages/Vote.js

```diff
@@ -164,16 +127,57 @@ class Vote extends Component {
 
 const mapStateToProps = (state, ownProps) => {
   const showId = ownProps.match.params.id
-  const { order = [], viewing = null } = state.judge.queues[showId] || {}
+  const { order = [], loadingVotes = true, loadingSubmissions = true } =
+    state.judge.queues[showId] || {}
+  let { on: submissionId } = queryString.parse(state.router.location.search)
+
+  // If this submissionId is not parsable as a base-10 integer, throw it out
+  if (isNaN(parseInt(submissionId, 10))) {
```

Comment:
This check is not needed, as invalid id's won't be found in `order` by the check below.

Additionally, having this check couples the frontend to the id mechanism on the backend - if we changed to a public facing random string like we mentioned, this would break.